spec/p2p/reactor-survey.md

```diff
@@ -0,0 +1,247 @@
+
+> The consensus properties and the consensus algorithm are clearly specified in the arXiv paper. In the this section we could thus start our discussions from the consensus gossip property that provides a clear abstraction. As this level of detail of specification is missing for the other reactors, we need to survey them a bit in the following in more detail
+
+## Mempool
+
+Considering the whole CometBFT system, intuitively, there should be end-to-end 
+requirements regarding transaction processing, e.g., "every transaction submitted
+should be eventually put into a block" or "every transaction should be put in
+at most one block". Note that these are just simple examples that are neither ensured
+by current implementation nor are actually required by the system (e.g., typically we 
+do not care about "all" transaction but rather "all valid"). 
+Which of such requirements are actually ensured by the 
+system depends on the guarantees by the concerned parts of the system.
+
+The mempool is a distributed pool of "pending" transactions.
+A pending transaction is a "valid" transaction that has been submitted by a
+client of the blockchain but has not yet been committed to the blockchain.
+The mempool is thus fed with client transactions,
+that a priori can be submitted to any node in the network.
+And it is consumed by the consensus protocol, more specifically by validator nodes,
+which retrieve from the mempool transactions to be included in proposed blocks.
+
+> Note that in this document, we want to capture (an overapproximation of) the requirements the mempool puts on p2p. The notions of "pending" and "valid" seem mostly to be conditions that the mempool can use to drop transactions (and exempt them from beeing spread over the network). Thus for the purpose of this document we will largely ignore them (they should eventually be addressed [here](https://github.com/cometbft/cometbft/issues/223)). 
+
+As 
+- full nodes add transactions to the mempool following client requests, and
+- validators read transactions from the mempool to compose proposed blocks,
+
+the mentioned end-to-end requirements on transactions translate into requirements 
+on the mempool, e.g., every transaction that is submitted to a full node should 
+eventually be present at the mempool of a validator/proposer. 
+
+For the purpose of this document (specifying what the mempool needs from p2p), 
+we do not strive to precisely state the end-to-end requirements. Rather, we 
+will show that the arguably strongest requirement on the mempool do translate roughly
+into requirements on the p2p layer that are similar to those of the
+consensus reactor.
+
+- end-to-end requirement: "every transaction submitted should be eventually put into a block"
```

Comment:
Not possible to guarantee.

Reply:
See [this comment](https://github.com/cometbft/cometbft/pull/80/files/f2f73d279306cc15b9460b1e217a82ace92fd09d#r1158159787).